### ISSUE TITLE
summary.txt number of lines changes

### ISSUE THREAD

**Issue by bernt-matthias** [fabio-cumbo/chopin#1]
We are running tests using the following command 
```
chopin2  --dataset 'iris.csv' --fieldsep ','  --dimensionality 1000 --levels 100 --retrain 10 --stop --crossv_k 5  --select_features --group_min 1 --accuracy_threshold 60.0 --accuracy_uncertainty_perc 5.0  --dump --cleanup --nproc 4 --verbose
```
using this test data https://github.com/galaxyproject/tools-iuc/blob/main/tools/chopin2/test-data/iris.csv

We observe that the number of output lines differs from run to run, i.e. sometimes there are  5 and sometimes 4 data lines (missing is `64e07b74f471e3b056c3e86b2b2af45d`). I would estimate that the last line is missing in approx 1/4 of the runs.

Is this expected?

**Comment by cumbof** [comment]
Hi @bernt-matthias, that’s definitely not expected.

I suppose you’re running the latest version of `chopin2`, right?
https://github.com/galaxyproject/tools-iuc/blob/ea1cf7593dd429e0f6272f9cf3d08cd3a46b6235/tools/chopin2/macros.xml#L2

If so, I’m going to investigate this ASAP

**Comment by bernt-matthias** [comment]
Exactly

**Comment by cumbof** [comment]
Sounds good. I’ll try to look at that later today and I’ll be back to you ASAP.
Thanks for reporting that issue

**Comment by cumbof** [comment]
@bernt-matthias, problem solved.

I’ve released [@cumbof/chopin2@1.0.9.post1](https://github.com/cumbof/chopin2/releases/tag/1.0.9.post1) which is already available in pypi and conda.
The tests in [chopin2.xml](https://github.com/galaxyproject/tools-iuc/blob/main/tools/chopin2/chopin2.xml) are still valid, but the [test-data](https://github.com/galaxyproject/tools-iuc/tree/main/tools/chopin2/test-data) may slightly change.
I’m going to submit a PR with the updated `summary*.tsv` and `selection*.tsv` files later today.

**Comment by bernt-matthias** [comment]
Excellent. Thanks
